### PR TITLE
Fix `redshift_data` hook to consider access and secret key from the connections object not from extra

### PR DESCRIPTION
**Pull request by bharanidharan14**
- Consider access and secret key if it is passed in connection login and password field
- Consider AWS session token if it is passed in `extra_config`

closes: #741

### CODE REVIEW

File: astronomer/providers/amazon/aws/hooks/redshift_data.py [hunk 1]
```diff
@@ -77,6 +77,9 @@ def get_conn_params(self) -> Dict[str, Union[str, int]]:
                 if "secret_access_key" in extra_config
                 else extra_config["aws_secret_access_key"]
             )
+        elif connection_object.login:
+            conn_params["aws_access_key_id"] = connection_object.login
+            conn_params["aws_secret_access_key"] = connection_object.password
```
Comment (pankajastro) [inline astronomer/providers/amazon/aws/hooks/redshift_data.py:82]: between this method andhttps://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/amazon/aws/hooks/base_aws.py#L33 there is lot of redundant code we should see if we can improve it

File: tests/amazon/aws/hooks/test_redshift_data.py [hunk 2]
```diff
@@ -248,9 +248,11 @@ def test_execute_query_exception(mock_conn, mock_params):
 
 
 @pytest.mark.parametrize(
-    "connection_details, expected_output",
+    "mock_login, mock_pwd, connection_details, expected_output",
     [
         (
+            "",
```
Comment (pankajastro) [inline tests/amazon/aws/hooks/test_redshift_data.py:254]: we should add some value empty does not make sense

File: tests/amazon/aws/hooks/test_redshift_data.py [hunk 3]
```diff
@@ -269,6 +271,8 @@ def test_execute_query_exception(mock_conn, mock_params):
             },
         ),
         (
+            "",
```
Comment (pankajastro) [inline tests/amazon/aws/hooks/test_redshift_data.py:274]: same here

File: tests/amazon/aws/hooks/test_redshift_data.py [hunk 4]
```diff
@@ -286,11 +290,31 @@ def test_execute_query_exception(mock_conn, mock_params):
                 "database": "",
             },
         ),
+        (
+            "test",
+            "test",
+            {
+                "db_user": "test_user",
+                "cluster_identifier": "",
+                "region": "",
```
Comment (pankajastro) [inline tests/amazon/aws/hooks/test_redshift_data.py:299]: same here

File: astronomer/providers/amazon/aws/hooks/redshift_data.py (hunk 1 above)
Comment (bharanidharan14) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:82]: Yes that's async hook which gets the connection info this `redshift_data` hook  class

Comment (bharanidharan14) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:82]: We can fix this redundant code

Comment (bharanidharan14) [reply astronomer/providers/amazon/aws/hooks/redshift_data.py:82]: created ticket https://app.zenhub.com/workspaces/airflow-providers-team-61a89ee7fbf048001dc8378d/issues/astronomer/astronomer-providers/747

**Comment by bharanidharan14** [comment]
@pankajastro @rajaths010494 Addressed your review comments, can you please take a look at it

**Review by pankajkoti — APPROVED** [review]

**Review by rajaths010494 — APPROVED** [review]

**Pull request merged**